### PR TITLE
feat: Implement CI Reporting for Playwright Tests

### PR DESCRIPTION
**Pull request by JoCa96**
- CI now generates HTML report, which can be downloaded and then shown using `npx playwright show-report`
- Merge blob-reports and screenshots for cleaner artifact overview

### CODE REVIEW

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -58,6 +58,17 @@ jobs:
...
 
+      - name: Upload Reports
+        uses: actions/upload-artifact@v4
+        if: ${{ !cancelled() }}
+        with:
+          name: blob-report-${{ matrix.shard }}
+          path: |
+            ./packages/**/blob-report
+            ./apps/**/blob-report
+            !./**/node_modules/**
```
Comment (larsrickert) [inline .github/workflows/playwright.yml:69]:
```suggestion
          path: |
            ./packages/*/blob-report
            ./apps/*/blob-report
```

Isn't this the same?

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -58,6 +58,17 @@ jobs:
           # prevent warnings if shard does not update any screenshots
           if-no-files-found: ignore
 
+      - name: Upload Reports
+        uses: actions/upload-artifact@v4
+        if: ${{ !cancelled() }}
```
Comment (larsrickert) [inline .github/workflows/playwright.yml:63]: Should we only upload the reports if the pipeline failed?

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -67,6 +78,56 @@ jobs:
...
+        run: find . -type d -name blob-report | xargs -n1 -I {} sh -c 'npx playwright merge-reports --reporter html {} && mv playwright-report {}/../playwright-report'
+
+      - name: Upload HTML report
+        uses: actions/upload-artifact@v4
+        with:
+          name: html-report--attempt-${{ github.run_attempt }}
+          path: |
+            ./packages/**/playwright-report
+            ./apps/**/playwright-report
+            !./**/node_modules/**
```
Comment (larsrickert) [inline .github/workflows/playwright.yml:128]:
```suggestion
            ./packages/*/playwright-report
            ./apps/*/playwright-report
```

File: packages/chartjs-plugin/playwright.config.ts [hunk 4]
```diff
@@ -18,7 +18,8 @@ export default defineConfig({
   fullyParallel: true,
   forbidOnly: !!process.env.CI, // fail build on CI if we left test.only in the source code
   retries: process.env.CI ? 2 : 0, // retry on CI only
-  reporter: [["html", { open: "never" }]],
+  /* In the CI pipeline it generates github and blob reports, locally only a html report is generated */
+  reporter: process.env.CI ? [["github"], ["blob"]] : [["html", { open: "never" }]],
```
Comment (larsrickert) [inline packages/chartjs-plugin/playwright.config.ts:22]: Do we need both blob and github or is one of them enough?

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/chartjs-plugin/playwright.config.ts (hunk 4 above)
Comment (larsrickert) [inline packages/chartjs-plugin/playwright.config.ts:22]: the CI run is spammed with Playwright comments, is this caused by the github reporter?
image

File: .github/workflows/playwright.yml (hunk 2 above)
Comment (JoCa96) [reply .github/workflows/playwright.yml:63]: Is there a benefit of not uploading the reports for successful runs?

Comment (larsrickert) [reply .github/workflows/playwright.yml:63]: I guess only saving disk space and keep the log minimal.
But I am also fine with keeping it like this

File: packages/chartjs-plugin/playwright.config.ts (hunk 4 above)
Comment (JoCa96) [reply packages/chartjs-plugin/playwright.config.ts:22]: Yes, I replaced it with the dot reporter, when running in the CI now.

Comment (larsrickert) [reply packages/chartjs-plugin/playwright.config.ts:22]: I think you changed it for the headless package only, chartjs-plugin and sit-onyx still have the github reporter

Comment (JoCa96) [reply packages/chartjs-plugin/playwright.config.ts]: done

**Comment by larsrickert** [comment]
Is it possible to not show the error when no screenshots are updated?
image

**Review by larsrickert — APPROVED** [review]

**Comment by JoCa96** [comment]
> Is it possible to not show the error when no screenshots are updated? image

No, AFAIK after 5 minutes of googling.

**Pull request merged**